### DISCUSSION TITLE
fluent bit deployed by docker  indicates no mechanism error when configuring kafka KERBEROS

### DISCUSSION THREAD

**Discussion by sailthesea** [fluent/fluent-bit#8328] [Q&A] [answered]
Startup error reported as follows：
 [error] [output:kafka:kafka.0] fluent-bit#producer-1: [thrd:app]: fluent-bit#producer-1: sasl_ssl://pxc1:9092/bootstrap: Failed to initialize SASL authentication: SASL handshake failed (start (-4)): SASL(-4): no mechanism available: No worthy mechs found (after 0ms in state AUTH_REQ)

I deployed fluent bit using docker：docker pull fluent/fluent-bit:latest-debug

Start command：docker run  -d --restart=always --network=host --name fluent-bit  -v /usr/vtm/backend/init/timezone:/etc/timezone:ro -v /usr/vtm/backend/init/localtime:/etc/localtime:ro -v /usr/vtm/etc/fluent-bit/:/fluent-bit/etc -v /var/vtm/log/:/fluent-bit/log/ fluent/fluent-bit:latest-debug

fluentbit configuration：
[OUTPUT]
    name    kafka
    format    json
    match    JSON-EVENT
    brokers    pxc1:9092
    topics    my-topic2
    rdkafka.sasl.mechanism GSSAPI
    rdkafka.sasl.kerberos.service.name kafka-server
    rdkafka.sasl.kerberos.principal kafka-client@ABC.COM
    rdkafka.sasl.kerberos.keytab /fluent-bit/etc/kerberos/zk-client.keytab
    rdkafka.security.protocol SASL_SSL
    rdkafka.ssl.ca.location /fluent-bit/etc/ssl/server.cert-signed
    rdkafka.ssl.certificate.location /fluent-bit/etc/ssl/ca-cert
    rdkafka.ssl.key.location      /fluent-bit/etc/ssl/ca-key
    rdkafka.ssl.key.password  test1234
    rdkafka.ssl.endpoint.identification.algorithm none

Special note:Messages can be successfully sent using confluent_kafka and using the latest librdkafka.Using fluent-bit source code deployment can also be successful.

Asking for help. Thank you

**Comment by patrick-stephens** [comment, marked as answer]
I imagine it needs to be built with the relevant library support and then these also need to be bundled in the production image. If it works locally with source builds you can compare those to figure out what needs changing for the container build.

We have to do similar things for some of the other runtime-only dependency options.
